src/state/opamEnv.ml

```diff
@@ -609,8 +658,14 @@ let display_setup root ~dot_profile shell =
         && not complete_zsh then
           not_set
         else ok in
-      [ ("init-script"     , Printf.sprintf "%s" pretty_init_file);
-        ("auto-completion" , completion);
+      let eval_env =
+        try
+          if bool_of_string @@ Sys.getenv "OPAM_NOEVAL_ENV" then ok else not_set
```

Comment:
`OpamStd.Config.env_bool "NOEVALENV"` (with the variable as `OPAMNOEVALENV` above)